### PR TITLE
Multizone VAV supply air temperature reset exceptions

### PR DESCRIPTION
**Pull request by leijerry888**
Addendum AP of 90.1 2016 further details exceptions for multi-zone VAV system SAT reset requirements. The additional exemptions are implemented as:

1. Differentiate exceptions between climate zones (0A, 1A and 3A) and (2A).
2. Add design OA requirements for these two groups of exceptions.
3. Add ERV and design OA ratio check for additional exception.

For more information check Section B.2.6 in the Energy Savings Analysis: [ANSI/ASHRAE/IES Standard 90.1-2019 report](https://www.energycodes.gov/sites/default/files/2021-07/20210407_Standard_90.1-2019_Determination_TSD.pdf).

### CODE REVIEW

File: lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb [hunk 1]
```diff
@@ -317,16 +317,60 @@ def air_loop_hvac_supply_air_temperature_reset_required?(air_loop_hvac, climate_
       return is_sat_reset_required
     end
 
+    # check if design outside air is less than 10,000cfm (5000L/s) 90.1 2019 6.5.3.5 Exception 1 and 2
+    design_oa_m3s = air_loop_hvac.sizingSystem.designOutdoorAirFlowRate.to_f
```
Comment (mdahlhausen) [inline lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb:321]: The designOutdoorAirFlowRate method returns an optional double.  And this value is often autosized, so not set on the system before the system sizing is run.  The subsequent less than check will fail for most models.  Likewise, it is unclear why the follow if < 0.0001 statement is necessary - should the floor for this value be zero anyways?

File: lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb [hunk 2]
```diff
@@ -317,16 +317,60 @@ def air_loop_hvac_supply_air_temperature_reset_required?(air_loop_hvac, climate_
...
+    if design_oa_m3s < 0.0001
+      design_oa_m3s = air_loop_hvac.sizingSystem.autosizedDesignOutdoorAirFlowRate.to_f
+    end
+    design_oa_cfm = OpenStudio.convert(design_oa_m3s, 'm^3/s', 'cfm').get
+
+    # check if there is erv 90.1 2019 Exceptions to 6.5.3.5 Exception 3
+    has_erv = air_loop_hvac_energy_recovery?(air_loop_hvac)
+    design_sa_m3s = air_loop_hvac.designSupplyAirFlowRate.to_f
```
Comment (mdahlhausen) [inline lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb:329]: same issue of the designSupplyAirFlowRate being an optional double.

File: lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb (hunk 1 above)
Comment (mdahlhausen) [reply lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb:321]: Refer to other HVAC controls methods that refer to the autosized values, such as `autosizedDesignSupplyAirFlowRate.is_initialized` in Standards.AirLoopHVAC.rb

**Review by mdahlhausen — CHANGES_REQUESTED** [review]
See comments in the code.  These methods need to account for autosized systems as well.

**Comment by mdahlhausen** [comment]
Also run rubocop on this file - seems like there may be a few uncaught errors.

File: lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb (hunk 1 above)
Comment (leijerry888) [reply lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb]: updated accordingly. 
(The original code was my improvised way of checking for autosized value, as if it is autosized, air_loop_hvac.sizingSystem.designOutdoorAirFlowRate.to_f gives 0, and I try not to put == between two floats, thus the 0.0001. Sorry for the confusions. Code is updated to same style as autosizedDesignSupplyAirFlowRate.is_initialized)

File: lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb (hunk 2 above)
Comment (leijerry888) [reply lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb]: Modified. I use an existing method from Standards.AirLoopHVAC.rb to get this value.

File: lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb (hunk 1 above)
Comment (leijerry888) [reply lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb]: > Also run rubocop on this file - seems like there may be a few uncaught errors.

I did not find rubocop errors in the code additions I made in this file. We could address rubocop problems in other parts in the big PR if you want.

**Review by mdahlhausen — APPROVED** [review]

**Pull request merged**